### DISCUSSION TITLE
Prevent multiple login for Openvpn client

### DISCUSSION THREAD

**Discussion by Emdad2001** [SoftEtherVPN/SoftEtherVPN#1812] [General]
I have a Softether VPN Server.for all account of a group, I want to limit multiple login to 1 user per account.
i did -> "maximum number of multiple logins = 1" but does not work for users use the same global IP Address
i think it must be the source IP when i change diffent ip address it work fine

**Comment by Mjzaret** [comment]
من هم چنین مشکلی دارم.
فقط زمانی که از دو آی پی پابلیک جدا بهش وصل میشم بهم اخطار میده ولی اگه از یک اینترنت مشترک بخواهیم وصل بشیم به صورت نامحدود میتونیم کانکت بشیم و هیچ تاثیری نداره. امیدوارم در آپدیت های بعدی این مشکل رو حل کنند

**Comment by shakibamoshiri** [comment]
it checks a source IP address not a user session  
You can write a script to periodically check for duplicate sessions and if there were, close all except the oldest one  
_And I think this feature wont be added to SE since it has not designed for selling VPN_

**Reply by chipitsine** [reply]
the feature will be added if it is useful and someone will submit PR

**Reply by shakibamoshiri** [reply]
Thanks for correcting me yet I think SE architecture is not suitable for selling VPNs to public. It lacks some essential features with which could be possible (e.g hooks, device per user, select interface/IP, manageable config file by admins, resource efficiency, etc)

**Reply by chipitsine** [reply]
if someone wants to contribute a PR with improving of architecture, it will be great

**Reply by Evengard** [reply]
The PR won't be even that hard, I think. I did a similar patch for PPP-based protocols, it was mostly trivial.

**Reply by Emdad2001** [reply]
do you have a script like that ?
